src/storage-client/src/controller/persist_handles.rs

```diff
@@ -81,6 +82,15 @@ impl<T: Timestamp + Lattice + Codec64> PersistReadWorker<T> {
                         PersistReadWorkerCmd::Update(id, since_handle) => {
                             since_handles.insert(id, since_handle).expect("PersistReadWorkerCmd::Update only valid for updating extant since handles");
                         }
+                        PersistReadWorkerCmd::DropHandle(id) => {
```

Comment:
Some people will argue that this should be done when we get a `Downgrade` to the empty frontier. But that might not be feasible here?

Reply:
That definitely happens when the read handle gets downgraded to the empty frontier; the thing I was trying to protect against is any kind of semantic drift in the future, i.e. we are saying that the persist worker simply cannot have the handle in memory once we cross this point. If that's more confusing than helpful, easy to remove.

Reply:
Not sure you changed this now, but I think it's ok to have the explicit `DropHandle`